tests/basic/vcluster_test.go

```diff
@@ -430,8 +432,460 @@ var _ = Describe("{VolumeDriverDownVCluster}", func() {
+		err = vc.CreateAndWaitVCluster()
+		log.FailOnError(err, "Failed to create VCluster")
+	})
+	It("Creates Nginx Deployment on Vcluster, Reboots a Px Cluster node, Validates Nginx Deployment", func() {
+		// Create Storage Class on Host Cluster
+		scName = fmt.Sprintf("nginx-app-sc-%v", time.Now().Unix())
+		err = CreateStorageClass(scName)
+		log.FailOnError(err, "Error creating Storageclass")
+		log.Infof("Successfully created StorageClass with name: %v", scName)
+		// Create PVC on VCluster
+		appNS = scName + "-ns"
+		pvcName, err = vc.CreatePVC("", scName, appNS, "")
+		log.FailOnError(err, fmt.Sprintf("Error creating PVC with Storageclass name %v", scName))
+		log.Infof("Successfully created PVC with name: %v", pvcName)
+		deploymentName := "nginx-deployment"
+		// Create Nginx Deployment on VCluster using the above PVC
+		err = vc.CreateNginxDeployment(pvcName, appNS, deploymentName)
+		log.FailOnError(err, "Error in creating Nginx Application")
+		log.Infof("Successfully created Nginx App on Vcluster")
+		log.Infof("Hard Sleep for 10 seconds after creation of Nginx Deployment")
+		time.Sleep(10 * time.Second)
+		// Validate if Nginx Deployment is healthy or not
+		err = vc.IsDeploymentHealthy(appNS, deploymentName, 1)
+		log.FailOnError(err, "Looks like Nginx Deployment is not healthy")
+		log.Infof("Nginx Deployment %s is healthy. Will kill Px and wait for its restart on all nodes now", deploymentName)
+		stepLog = "Reboot All Nodes one by one in rolling fashion"
+		Step(stepLog, func() {
+			nodesToReboot := node.GetWorkerNodes()
+			for _, n := range nodesToReboot {
+				log.InfoD("reboot node: %s", n.Name)
+				err = Inst().N.RebootNode(n, node.RebootNodeOpts{
+					Force: true,
+					ConnectionOpts: node.ConnectionOpts{
+						Timeout:         defaultCommandTimeout,
+						TimeBeforeRetry: defaultCommandRetry,
+					},
+				})
+				log.FailOnError(err, "Error while rebooting nodes")
+				log.Infof("wait for node: %s to be back up", n.Name)
+				err = Inst().N.TestConnection(n, node.ConnectionOpts{
+					Timeout:         defaultTestConnectionTimeout,
+					TimeBeforeRetry: defaultWaitRebootRetry,
+				})
+				if err != nil {
+					log.FailOnError(err, "Error while testing node status %v, err: %v", n.Name, err.Error())
+				}
+				log.FailOnError(err, "Error while testing connection")
+			}
+		})
+		// Validate if Nginx Deployment is healthy or not
+		err = vc.IsDeploymentHealthy(appNS, deploymentName, 1)
+		log.FailOnError(err, "Looks like Nginx Deployment is not healthy")
+		log.Infof("Nginx Deployment %s is healthy. Will kill Px and wait for its restart on all nodes now", deploymentName)
+	})
+	JustAfterEach(func() {
+		// VCluster, StorageClass and Namespace cleanup
+		err := vc.VClusterCleanup(scName)
+		if err != nil {
+			log.Errorf("Problem in Cleanup: %v", err)
+		} else {
+			log.Infof("Cleanup successfully done.")
+		}
+	})
+})
+
+// This Test Case is failing due to : https://portworx.atlassian.net/browse/PWX-34792
```

Comment:
Can we modify this tests to have multiple snapshots, suppose 5 and randomly selecting one for restore.

Reply:
Noted. Will add another test for this. The flow for this test case is meant to take latest snapshot.